bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api/ShellyHttpApi.java

```diff
@@ -471,15 +471,21 @@ private String request(String uri) throws IOException {
         try {
             result = innerRequest(uri);
         } catch (IOException e) {
+            String message = e.getMessage();
             String type = StringUtils.substringAfterLast(e.getCause().toString(), ".");
-            if (e.getMessage().contains("Timeout") || type.toLowerCase().contains("timeout")
-                    || e.getMessage().contains("Connection reset")) {
-                logger.debug("{}: Shelly API timeout ({}), retry", thingName, type);
+            if ((message != null) && (type != null)) {
```

Comment:
Extraneous parens

Reply:
Thanks for bringing this up. I changed a occurrences where e.getMessage() is used to check on null and map this to ""